pkg/node/client.go

```diff
@@ -0,0 +1,165 @@
+package node
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/kubernetes-sigs/container-object-storage-interface-api/apis/objectstorage.k8s.io/v1alpha1"
+	cs "github.com/kubernetes-sigs/container-object-storage-interface-api/clientset/typed/objectstorage.k8s.io/v1alpha1"
+)
+
+type NodeClient struct {
+	cosiClient *cs.ObjectstorageV1alpha1Client
+	kubeClient kubernetes.Interface
+}
+
+func NewClient() (*NodeClient, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := cs.NewForConfigOrDie(config)
+	kube := kubernetes.NewForConfigOrDie(config)
+	return &NodeClient{
+		cosiClient: client,
+		kubeClient: kube,
+	}, nil
+}
+
+const (
+	barNameKey      = "bar-name"
+	barNamespaceKey = "bar-namespace"
+)
+
+func parseValue(key string, ctx map[string]string) (string, error) {
```

Comment:
typically, not always, when a var is named `ctx` it is of golang type Context.

Reply:
makes sense, I renamed it